docs/std-encoding.md

```diff
@@ -26,15 +26,27 @@ Usage:
 |:-|:-|
 | `//encoding.json.decode('{"hi": "abc", "hello": 123}')` | `{'hello': 123, 'hi': (s: 'abc')}` |
 
-## `//encoding.proto.decode(protoFileDescriptor <: bytes, rootMessageName <: string, protobufMessage <: bytes) <: tuple`
+## `//encoding.proto.descriptor(protobufDefinition <: bytes) <: tuple`
 
-This method accepts [protobuf message](https://github.com/protocolbuffers/protobuf) information and data, and transfroms them to a built-in arr.ai value.
+This method accepts [profobuf](https://github.com/protocolbuffers/protobuf) `.proto` binay file and return a tuple which has [`FileDescriptorSet`](https://pkg.go.dev/google.golang.org/protobuf@v1.25.0/types/descriptorpb?tab=doc#FileDescriptorSet) describes protobuf message types in the binary file.
+
+For exmple:
+
+```arrai
+//encoding.proto.descriptor(//os.file('sys.pb'))
+```
+
+References: [sysl.pb](https://github.com/arr-ai/arrai/blob/master/translate/pb/test/sysl.pb)
+
+## `//encoding.proto.decode(descriptor <: tuple, messageTypeName <: string, messageBytes <: bytes) <: tuple`
+
+This method accepts [protobuf message](https://github.com/protocolbuffers/protobuf) information and data, and transfroms them to a built-in arr.ai value tuple.
```

Comment:
```suggestion
This method accepts three parameters:

- a tuple representation of a [`FileDescriptorSet`](https://pkg.go.dev/google.golang.org/protobuf@v1.25.0/types/descriptorpb?tab=doc#FileDescriptorSet) (as produced by `//encoding.proto.descriptor`).
- the name of the message to be decoded.
- the content of an encoded [protobuf message](https://github.com/protocolbuffers/protobuf).

It returns a tuple representation of the encoded message.
```

Reply:
It would also be good to describe the differences to a proto (e.g. the `a`, `s`, `i`, etc. props)

Reply:
I think it is better to focus on the description of this method as `a`, `s` and `i` is protobuf knowledges, and reader can get them more separately.